### PR TITLE
fix: @trpc/client has undefined global #2529 (globalThis only)

### LINKED ISSUE #2493: bug: v10 @trpc/client has undefined `global` in workers
### Provide environment information

```
  System:
    OS: macOS 12.5
    CPU: (8) arm64 Apple M1
    Memory: 158.41 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.5.0 - ~/.nvm/versions/node/v18.5.0/bin/node
    Yarn: 1.22.17 - ~/.yarn/bin/yarn
    npm: 8.12.1 - ~/.nvm/versions/node/v18.5.0/bin/npm
  Browsers:
    Chrome: 104.0.5112.101
    Chrome Canary: 106.0.5249.0
    Firefox Developer Edition: 104.0
    Safari: 15.6
  npmPackages:
    @trpc/client: 10.0.0-alpha.47 => 10.0.0-alpha.47 
    @trpc/server: =10.0.0-alpha.47 => 10.0.0-alpha.47 
```

### Describe the bug

When attempting to deploy my app I was getting the following error:
```
✘ [ERROR] A request to the Cloudflare API (/accounts/000000011111111122222222/workers/scripts/app-edge) failed.

  Uncaught ReferenceError: global is not defined
    at worker.js:10:10
   [code: 10021]

  
  If you think this is a bug, please open an issue at:
  https://github.com/cloudflare/wrangler2/issues/new/choose
```

### To reproduce

Try to deploy something using something akin to my gist for [wrapping Durable Objects in TRPC](https://gist.github.com/colelawrence/ba912db3cd1539ee820c3fe5e67654df)

### Additional information

I was able to fix in the short term by adding a `banner` via esbuild: `if (typeof global === "undefined") { globalThis.global = globalThis; }`

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR fixing this bug!

### PR DESCRIPTION
**Pull request by colelawrence**
For more information about `globalThis`, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis

Closes #2493

Expands on #2529

## 🎯 Changes

What changes are made in this PR? Is it a feature or a bug fix?

## ✅ Checklist

- [ ] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: packages/client/src/internals/fetchHelpers.ts [hunk 1]
```diff
@@ -2,11 +2,10 @@ export function getWindow() {
   if (typeof window !== 'undefined') {
```
Comment (sachinraja) [inline packages/client/src/internals/fetchHelpers.ts:2]: Can we remove this one too?

Comment (colelawrence) [reply packages/client/src/internals/fetchHelpers.ts:2]: I tried getting rid of getWindow() all together, but there seemed to be an issue with `jest-environment-jsdom` not setting fetch on `globalThis` properly.
https://github.com/trpc/trpc/runs/8041790807?check_suite_focus=true#step:5:710

Comment (colelawrence) [reply packages/client/src/internals/fetchHelpers.ts:2]: Oh wait, no you're right. It's the tests that are wrong
https://github.com/trpc/trpc/blob/next/packages/server/test/interop/clientInternals.test.ts#L42-L45

Comment (colelawrence) [reply packages/client/src/internals/fetchHelpers.ts:2]: I'd love some help fixing those tests. Then everywhere we currently call `getWindow()` can just be replaced by `globalThis`.

**Review by KATT — APPROVED** [review]

File: packages/client/src/internals/fetchHelpers.ts (hunk 1 above)
Comment (sachinraja) [reply packages/client/src/internals/fetchHelpers.ts:2]: Shouldn't it still work if we add the property to `global`?

**Pull request merged**